go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go

```diff
@@ -260,6 +205,261 @@ func TestSchemaChange(t *testing.T) {
+			expandedColumnNames: `t1,t2,t3,t5,t6`,
+		},
+		{
+			name:                "expanded: character sets",
+			fromSchema:          `id int primary key, c1 char(3) charset utf8, c2 char(3) charset utf8mb4, c3 char(3) charset ascii, c4 char(3) charset utf8mb4, c5 char(3) charset utf8, c6 char(3) charset latin1`,
+			toSchema:            `id int primary key, c1 char(3) charset utf8mb4, c2 char(3) charset utf8, c3 char(3) charset utf8, c4 char(3) charset ascii, c5 char(3) charset utf8, c6 char(3) charset utf8mb4`,
+			expandedColumnNames: `c1,c3,c6`,
+		},
+		{
+			name:                "expanded: enum",
+			fromSchema:          `id int primary key, e1 enum('a', 'b'), e2 enum('a', 'b'), e3 enum('a', 'b'), e4 enum('a', 'b'), e5 enum('a', 'b'), e6 enum('a', 'b'), e7 enum('a', 'b'), e8 enum('a', 'b')`,
+			toSchema:            `id int primary key, e1 enum('a', 'b'), e2 enum('a'), e3 enum('a', 'b', 'c'), e4 enum('a', 'x'), e5 enum('a', 'x', 'b'), e6 enum('b'), e7 varchar(1), e8 tinyint`,
+			expandedColumnNames: `e3,e4,e5,e6,e7,e8`,
+		},
+		{
+			name:                "expanded: set",
+			fromSchema:          `id int primary key, e1 set('a', 'b'), e2 set('a', 'b'), e3 set('a', 'b'), e4 set('a', 'b'), e5 set('a', 'b'), e6 set('a', 'b'), e7 set('a', 'b'), e8 set('a', 'b')`,
+			toSchema:            `id int primary key, e1 set('a', 'b'), e2 set('a'), e3 set('a', 'b', 'c'), e4 set('a', 'x'), e5 set('a', 'x', 'b'), e6 set('b'), e7 varchar(1), e8 tinyint`,
+			expandedColumnNames: `e3,e4,e5,e6,e7,e8`,
+		},
+	}
+
+	var (
+		createTableWrapper = `CREATE TABLE onlineddl_test(%s)`
+		dropTableStatement = `
+			DROP TABLE onlineddl_test
+		`
+		tableName   = "onlineddl_test"
+		ddlStrategy = "online -declarative -allow-zero-in-date"
+	)
+
+	removeBackticks := func(s string) string {
+		return strings.Replace(s, "`", "", -1)
+	}
+
+	for _, testcase := range testCases {
+		t.Run(testcase.name, func(t *testing.T) {
+
+			// func testOnlineDDLStatement(t *testing.T, alterStatement string, ddlStrategy string, executeStrategy string, expectHint string, expectError string, skipWait bool) (uuid string) {
+			// func testOnlineDDLStatement(t *testing.T, alterStatement string, ddlStrategy string, executeStrategy string, tableName string, expectHint string) (uuid string) {
+
+			t.Run("ensure table dropped", func(t *testing.T) {
+				uuid := testOnlineDDLStatement(t, dropTableStatement, ddlStrategy, "vtgate", tableName, "")
+				onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+				checkTable(t, tableName, false)
+			})
+
+			t.Run("create from-table", func(t *testing.T) {
+				fromStatement := fmt.Sprintf(createTableWrapper, testcase.fromSchema)
+				uuid := testOnlineDDLStatement(t, fromStatement, ddlStrategy, "vtgate", tableName, "")
+				onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+				checkTable(t, tableName, true)
+			})
```

Comment:
It's not obvious to me how these are confirming it's revertible, but assuming it is (you would certainly know).

Reply:
Ah. This is just preparation. Per test case, we (re)create the table, then run a migration, then see if the migration is revertible. Added comments to clarify.